### DISCUSSION TITLE
Allow configuration settings to be set via environment variables

### DISCUSSION THREAD

**Discussion by AtomBaf** [duckdb/duckdb#9881] [Ideas]
First, thx for this great product.

Today when setting up an minio S3 access, in my python code I have to setup something like this: (I'm in a corporate environment)
```python
duckdb.query('SET custom_extension_repository="http://my_custom_location/duckdb/extensions"')
duckdb.query("SET s3_url_style='path'")
duckdb.query(f"SET s3_endpoint='{endpoint}'")
duckdb.query("SET s3_use_ssl=false")
duckdb.query(f"SET s3_access_key_id='{key}'")
duckdb.query(f"SET s3_secret_access_key='{secret}'")
duckdb.query('LOAD httpfs')
```

This is something that I have to repeat each time.

I would like to be able to set these settings once for all with environment variables. For instance with: `DUCKDB_<uppercase_name_of_the_configuration_variable>` 

This way my code will look much cleaner

**Comment by akode** [comment]
This is at least partially possible.
s3_access_key_id can be set using environment variable AWS_ACCESS_KEY_ID
s3_secret_access_key can be set using environment variable AWS_SECRET_ACCESS_KEY

See https://duckdb.org/docs/stable/core_extensions/httpfs/s3api#credential_chain-provider for more details about using the env provider for setting secrets.

**Reply by agonencyto** [reply]
the problem is with s3_url_style it should be expose also as DUCKDB_S3_USE_SSL
